### ISSUE TITLE
Select: Accessibility failure due to label

### ISSUE THREAD

**Issue by Joseph-Hitchcock** [IBM/carbon-components-react#1131]
## Detailed description

> Describe in detail the issue you're having.

When using a Select component, the `<label>` element is being rendered after the `<select>` element, causing the following accessibility error: `Text inputs and select elements must have a label before the input control.`

> Is this a feature request (new component, new icon), a bug, or a general issue?

bug

> Is this issue related to a specific component?

Select

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

6.14.3

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

IBM Cloud

**Comment by joshblack** [comment]
Hi @Joseph-Hitchcock! Thanks so much for taking the time to make this issue 😄 

With respect to the violation:

> Text inputs and select elements must have a label before the input control.

This is the first time that I've encountered this, as there is an assumption that if the `id` on the input node matches the `for` for the label then everything should work as expected. Would love to hear more about this technique to see where this requirement comes up for text and select elements 👍

**Comment by Joseph-Hitchcock** [comment]
Hi @joshblack! Here's a little bit more information on the accessibility issues we're facing.

When we run our accessibility script (required to pass builds), we get failures for every page with a select element in them, with errors like this one: 
![image](https://user-images.githubusercontent.com/40905872/43412519-592734ea-93e2-11e8-8f72-1d69749f2e82.png)

Using the IBM Dynamic Accessibility Plugin, we see the same errors like this:
![image](https://user-images.githubusercontent.com/40905872/43412540-6e8ba8ac-93e2-11e8-8fe7-0ed75dcc6fe2.png)

Inspecting our HTML, this is how the block in question appears:
![image](https://user-images.githubusercontent.com/40905872/43412574-870d6672-93e2-11e8-9f24-47d03c7836ed.png)
which seems like it satisfies that the `for` of the label matches the `id` of the input, but it's still failing for some reason.

I noticed that editing the element as HTML in the browser and cutting/pasting the `label` before the `input` fixed each individual violation, which was why @zgrossbart suggested that change in [his PR](https://github.com/IBM/carbon-components-react/pull/1134).
